### DISCUSSION TITLE
text().as_double() willing make floting number problem

### DISCUSSION THREAD

**Discussion by zhu1035167865** [zeux/pugixml#491] [Q&A]
SimulationNumericalParams_node.child("EndDay") = "111111111"

after I run the line below:
(SimulationNumericalParams_node.child("EndDay").first_child().text().as_double());

it will show that 1.1111111e+7

Is that possible keep it in 111111111?

I need to do Ten-digit high-precision calculation.

**Comment by zeux** [comment]
Your debugger probably just displays a truncated number, but the original number should be fully precise.

E.g. this code prints `111111111.000000 1`:

```
#include "pugixml.hpp"
#include <stdio.h>

int main()
{
        pugi::xml_document doc;
        doc.load_string("<snp><EndDay/></snp>");
        pugi::xml_node SimulationNumericalParams_node = doc.child("snp");
        SimulationNumericalParams_node.child("EndDay").text() = "111111111";
        double v = SimulationNumericalParams_node.child("EndDay").text().as_double();

        printf("%f %d\n", v, v == 111111111);
}
```

**Reply by zhu1035167865** [reply]
Assuming my "EndDay" number is 11000001
Assuming there is a "startday" that is 11000000,
When I try         double v = SimulationNumericalParams_node.child("EndDay").text().as_double() - SimulationNumericalParams_node.child("StartDay").text().as_double();

There result is 0, But it should be 1.

Am I missing some part?

That make me confused.

**Reply by zeux** [reply]
This program prints 1 for me, and really if it prints 0 this must mean some issue in the compiler or CRT, because as_double simply calls strtod in pugixml.

```
#include "pugixml.hpp"
#include <stdio.h>

int main()
{
        pugi::xml_document doc;
        doc.load_string("<snp><StartDay>11000000</StartDay><EndDay>11000001</EndDay></snp>");
        pugi::xml_node SimulationNumericalParams_node = doc.child("snp");
        double v = SimulationNumericalParams_node.child("EndDay").text().as_double() - SimulationNumericalParams_node.child("StartDay").text().as_double();
        printf("%f\n", v);
}
```
